backend/src/main/java/de/laura/project/service/TriviaService.java

```diff
@@ -14,26 +16,76 @@
 public class TriviaService {
 
     private final TriviaApiService triviaApiService;
+    private final TempTriviaQuestionDB tempTriviaQuestionDB;
+    private final AnswerRandomizerService answerRandomizerService;
+
+
+
 
     @Autowired
-    public TriviaService(TriviaApiService triviaApiService) {
+
+    public TriviaService(TriviaApiService triviaApiService, TempTriviaQuestionDB tempTriviaQuestionDB, AnswerRandomizerService answerRandomizerService) {
         this.triviaApiService = triviaApiService;
+        this.tempTriviaQuestionDB = tempTriviaQuestionDB;
+        this.answerRandomizerService = answerRandomizerService;
     }
 
-    public List<TriviaQuestionSet> getListDataPoints() {
 
-        List<TriviaApiData> triviaApiDataList = triviaApiService.getDataPoints().getTriviaApiData();
 
-        return triviaApiDataList.stream()
+
+    public List<TriviaQuestionSet> callQuestionList(int amount, int category, String difficulty) {
+
+        List<TriviaApiData> triviaApiDataList = triviaApiService.callQuestionList(amount, category, difficulty);
+
+        List<TriviaQuestionSet> triviaQuestionSetList = new ArrayList<>();
+
+        triviaQuestionSetList = triviaApiDataList.stream()
                 .map(triviaApiData -> TriviaQuestionSet.builder()
-                        .category(triviaApiData.getCategory())
-                        .type(triviaApiData.getType())
-                        .difficulty(triviaApiData.getDifficulty())
                         .question(triviaApiData.getQuestion())
                         .correct_answer(triviaApiData.getCorrect_answer())
-                        .incorrect_answers(triviaApiData.getIncorrect_answers())
+                        .answers(determineAnswers(triviaApiData))
                         .build()).collect(Collectors.toList());
 
+        addIDtoQuestionSet(triviaQuestionSetList);
+
+        tempTriviaQuestionDB.returnQuestionList(triviaQuestionSetList);
+
+        return triviaQuestionSetList;
+
+    }
+
+    private List<String> determineAnswers(TriviaApiData triviaApiData) {
+
+        ArrayList<String> answers = new ArrayList<String>();
+
+        int number = answerRandomizerService.generateRandomNumber();
+
+        for (int i = 0; i < 4; i++) {
+
+            if (i < number) {
+                answers.add(triviaApiData.getIncorrect_answers().get(i));
+            } else if (i == number) {
+                answers.add(triviaApiData.getCorrect_answer());
+            } else {
+                answers.add(triviaApiData.getIncorrect_answers().get(i - 1));
+            }
+        }
+
+        return answers;
+    }
+
+
+    public void addIDtoQuestionSet(List<TriviaQuestionSet> triviaQuestionSetList) {
+
+        int n = 1;
+
+        for (TriviaQuestionSet triviaQuestion : triviaQuestionSetList) {
+            triviaQuestion.setId(n);
+            n = n + 1;
+
+        };
```

Comment:
if you use this you can also use an indexed for
```suggestion
      
        for (int i = 0; i< triviaQuestionSetList.length ; i++) {
            triviaQuestionSetList[i].setId(i+1);
        };
```